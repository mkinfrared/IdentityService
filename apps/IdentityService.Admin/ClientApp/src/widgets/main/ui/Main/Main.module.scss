.Main {
  background-color: chocolate;
  display: flex;
  flex: 1;
  width: 100%;

  .sidebar {
    flex: 0 0 300px;
  }

  .page {
    flex: 1;
  }
}
