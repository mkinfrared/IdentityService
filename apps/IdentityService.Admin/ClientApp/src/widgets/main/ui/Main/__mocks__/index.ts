export { default } from "./Main.mock";
