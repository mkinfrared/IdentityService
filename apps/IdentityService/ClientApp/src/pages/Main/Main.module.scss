@import "../../styles/common";

.Main {
  @extend %loginRegisterPage;

  .formContainer {
    justify-content: flex-start;
  }
}

.Story {
}
