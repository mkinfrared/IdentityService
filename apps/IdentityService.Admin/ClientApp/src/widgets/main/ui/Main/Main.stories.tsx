import { Meta, Story } from "@storybook/react";

import Main from "./Main";
import { MainProps } from "./Main.type";
import css from "./Story.module.scss";

export default {
  title: "COMPONENTS/Main",
  component: Main,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<MainProps> = (args) => (
  <div className={css.Story}>
    <Main {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
