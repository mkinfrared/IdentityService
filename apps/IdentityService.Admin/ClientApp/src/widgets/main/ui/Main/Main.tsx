import { classNames } from "@identity-service/core";

import css from "./Main.module.scss";
import { MainProps } from "./Main.type";

const Main = ({ className, children }: MainProps) => (
  <main className={classNames(css.Main, className)} data-testid="Main">
    <section className={css.page}>{children}</section>
  </main>
);

export default Main;
