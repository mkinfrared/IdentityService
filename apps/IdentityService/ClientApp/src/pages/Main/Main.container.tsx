import { memo } from "react";
import { useLocation } from "react-use";

import getRoutePath from "utils/getRoutePath";

import MainFC from "./Main";
import { MainProps } from "./Main.type";

const Main = (props: Omit<MainProps, "path">) => {
  const { pathname } = useLocation();

  if (!pathname) {
    return null;
  }

  const path = getRoutePath(pathname);

  return <MainFC {...props} path={path} />;
};

export default memo(Main);
