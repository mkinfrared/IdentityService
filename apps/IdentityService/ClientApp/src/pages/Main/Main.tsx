import { classNames, lazyImport } from "@identity-service/core";
import { Card, Spinner } from "@identity-service/ui";
import { Suspense, memo } from "react";

import { useReturnUrl } from "hooks";

import css from "./Main.module.scss";
import { MainProps } from "./Main.type";

const ConsentForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "ConfirmEmail" */
      /* webpackPrefetch: true */
      "components/ConsentForm"
    ),
);

const ConfirmEmail = lazyImport(
  () =>
    import(
      /* webpackChunkName: "ConfirmEmail" */
      /* webpackPrefetch: true */
      "components/ConfirmEmailForm"
    ),
);

const LoginForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "LoginForm" */
      /* webpackPrefetch: true */
      "components/LoginForm"
    ),
);

const RegisterForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "RegisterForm" */
      /* webpackPrefetch: true */
      "components/RegisterForm"
    ),
);

const Main = ({ className, path }: MainProps) => {
  const returnUrl = useReturnUrl();

  let child: JSX.Element;

  switch (path) {
    case "login":
      child = <LoginForm className={css.formContainer} returnUrl={returnUrl} />;

      break;
    case "register":
      child = (
        <RegisterForm className={css.formContainer} returnUrl={returnUrl} />
      );

      break;
    case "confirmEmail":
      child = (
        <ConfirmEmail className={css.formContainer} returnUrl={returnUrl} />
      );

      break;

    case "consent":
      child = (
        <ConsentForm className={css.formContainer} returnUrl={returnUrl} />
      );

      break;

    default:
      child = <LoginForm className={css.formContainer} returnUrl={returnUrl} />;
  }

  return (
    <div className={classNames(css.Main, className)}>
      <Card className={css.card}>
        <Suspense
          fallback={<Spinner className={css.spinner} type="infinity" />}
        >
          {child}
        </Suspense>
      </Card>
    </div>
  );
};

export { Main };

export default memo(Main);

/* webpackInclude: /\.json$/ */
/* webpackExclude: /\.noimport\.json$/ */
/* webpackChunkName: "my-chunk-name" */
/* webpackMode: "lazy" */
/* webpackPrefetch: true */
/* webpackPreload: true */
