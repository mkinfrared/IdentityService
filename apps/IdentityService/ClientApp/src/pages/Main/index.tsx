import "@identity-service/ui/lib/esm/styles.min.css";
import "index.scss";
import ReactDOM from "react-dom";

import Main from "./Main.container";

ReactDOM.render(<Main />, document.getElementById("app"));
