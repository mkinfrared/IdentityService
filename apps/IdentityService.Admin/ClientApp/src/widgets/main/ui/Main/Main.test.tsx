import { render } from "@testing-library/react";

import Main from "./Main";

describe("<Main />", () => {
  const Component = <Main />;

  it("should be defined", () => {
    expect(Main).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Main");

    expect(element).toBeDefined();
  });
});
