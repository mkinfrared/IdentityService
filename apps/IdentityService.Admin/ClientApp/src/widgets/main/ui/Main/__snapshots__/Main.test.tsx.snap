// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match the snapshot 1`] = `
<div>
  <main
    class="Main"
    data-testid="Main"
  >
    <section
      class="page"
    />
  </main>
</div>
`;
